Decouple user preferences, attributes, and flags from getOption()

drsnyder opened the pull request:
This work decouples preferences, attributes, and flags from `User::getOption()`.  See the [ticket](https://wikia-inc.atlassian.net/browse/SERVICES-469) for the details and rationale. 

TODO:
- [x] create the setters for each
- [x] replace the `getOption` call sites. We are using [this spreadsheet](https://docs.google.com/spreadsheets/d/11B8FupwtaahpE2n9GvYeaSMM43HDhmWM-VoCh1B572A/edit#gid=0) to coordinate and track that effort
- [x] replace the `setOption` call sites. Tracking TBD
- [x] add logging to `get`/`set` option so we can log when we miss any

/cc @Wikia/services-team

nmonterroso commented:
As part of this, should we also do something like
```php
public function setOption(...) {
   log("WHY ARE YOU CALLING SETOPTION? YOU CALLED IT FROM <location in code>");
  $this->setOptionForRealsies(...);
}

private function setOptionForRealsies(...) {
  // logic that used to be in setOption
}
```
That way we can track down set/get calls that are dynamically invoked using `call_user_func` or we missed in our grep. OTW LGTM.

drsnyder committed:
getOption changes in extensions/wikia/AchievementsII/services/AchAwardingService.class.php

hidepersonalachievements

drsnyder committed:
getOption changes in extensions/wikia/AchievementsII/services/AchUserProfileService.class.php

hidepersonalachievements

drsnyder committed:
getOption changes in extensions/wikia/AdEngine/AdEngine2PageTypeService.class.php

showAds

drsnyder committed:
getOption changes in extensions/wikia/Answers/AttributionCache.class.php

hidefromattribution

drsnyder committed:
getOption changes in extensions/wikia/Answers/DefaultQuestion.php

watchcreations

drsnyder committed:
getOption changes in extensions/wikia/ArticleComments/classes/ArticleComment.class.php

watchdefault

drsnyder committed:
getOption changes in extensions/wikia/AutoFollow/AutoFollow.hooks.php

language,marketingallowed

drsnyder committed:
getOption changes in extensions/wikia/AutoFollow/AutoFollowTask.class.php

language,autowatched-already

Commit:
updating getOption calls in includes/specials/SpecialMovepage.php

tiffperumpail committed:
Changed WikiaMiniUpload_body getOption to getGlobalPreference

tiffperumpail committed:
Changed getOption to getGlocaPreference in WikiaMobileFooterService

drsnyder committed:
Fill in setLocalPreference

drsnyder committed:
getOption changes in extensions/wikia/AutomaticWikiAdoption/AutomaticWikiAdoptionHelper.class.php

LastAdoptionDate,adoptionmails-$wgCityId

drsnyder committed:
getOption changes in extensions/wikia/AutomaticWikiAdoption/maintenance/AutomaticWikiAdoptionGatherData.php

adoptionemails-$wgCityId

drsnyder committed:
getOption changes in extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php

watchdefault,watchcreations,minordefault

drsnyder committed:
getOption changes in extensions/wikia/CategoryExhibition/CategoryExhibitionSection.class.php

CategoryExhibitionSortType,CategoryExhibitionDisplayType

tiffperumpail committed:
Changed getOption in WikiaPhotoGalleryHelper to getGlobalPreference

tiffperumpail committed:
Merge branch 'SERVICES-469' of https://github.com/Wikia/app into SERVICES-469

tiffperumpail committed:
Changed getOption in WikiaApiAjaxLogin to getGlobalFlag

tiffperumpail committed:
Changed getOption to getGlobalPreference in WidgetShoutBox

tiffperumpail committed:
Changed getOption to getGlobalPreference in commented out code block in WidgetAnswers

tiffperumpail committed:
Changed getBoolOption to getGlobalPreferences in WatchSubPagesHelper

tiffperumpail committed:
Change getOption to getGlobalFlag in WallMessage.class

drsnyder committed:
getOption changes in extensions/wikia/CategorySelect/CategorySelectController.class.php

showhiddencats

tiffperumpail committed:
Change getOption to getGlobalPreference for wall_sort_

tiffperumpail committed:
Change getGlobalFlag to getGlobalPreference for wallshowsource

tiffperumpail committed:
Changed getGlobalFlag to getGlobalPreference for walldelete

tiffperumpail committed:
Changed getOption to getGlobalPreference for widget_bookmark_pages

tiffperumpail committed:
Change getOption to getGlobalPreference for wall_sort_

tiffperumpail committed:
Change getOption to getGlobalFlag for 'request-rename' and 'wasRenamed'

tiffperumpail committed:
Changed getOption to getGlobalAttribute for renameDate

tiffperumpail committed:
Changed getOption to getGlobalAttribute for avatar

tiffperumpail committed:
Changed getOption to getGlobalFlag for disabled

tiffperumpail committed:
Changed getOption to getGlobalFlag for disabled; Changed getOption to getGlobalAttribute foreach  in array of attributes; Changed getOption to getGlobalFlag for USER_EDITED_MASTERHEAD_PROPERTY .  and for USER_EVER_EDITED_MASTHEAD

tiffperumpail committed:
Changed getOption to getGlobalAttribute for AVATAR_USER_OPTION_NAME constant, which is set to 'avatar'

tiffperumpail committed:
Changed getOption to getGlobalPreference for -toolbar-contents

tiffperumpail committed:
Changed getOption to getGlobalPreference for unsubscribed

tiffperumpail committed:
Changed getOption to getGlobalFlag for NotConfirmedSignup;
Changed getOption to getGlobalAttribute for SignUpRedirect;
Changed getOption to getGlobalAttribute for new_email

tiffperumpail committed:
Changed getOption to getGlobalAttribute for SignUpRedirect; Changed getOption to getGlobalFlag for NotConfirmedSignup; Changed getOption to getGlobalAttribute for new_email

tiffperumpail committed:
Changed getOption to getGlobalAttribute for NOT_CONFIRMED_LOGIN_OPTION_NAME constant set to NotConfirmedLogin

tiffperumpail committed:
Changed getOption to getGlobalAttribute for new_email

tiffperumpail committed:
Changed getOption to getGlobalPreference for language; Changed getOption to getGlobalFlag for NotConfirmedSignup; Changed getOption to getGlobalFlag for cr_mailed; Changed getOption to getGlobalAttribute for new_email

tiffperumpail committed:
Changed getOption to getGlobalAttribute for NotConfirmedSignUp

tiffperumpail committed:
Changed getOption to getGlobalPreference for rememberpassword

tiffperumpail committed:
Changed getOption to getGlobalPreference for unsubscribed

tiffperumpail committed:
Changed getBoolOption to getGlobalPreference for unsubscribed in commented out code block

tiffperumpail committed:
Change getBoolOption to getGlobalPreference for minordefault

tiffperumpail committed:
Change getOption to getGlobalPreference for language and getOption to getGlobalFlag for wasRenamed

tiffperumpail committed:
Change getOption to getGlobalPreference for language

tiffperumpail committed:
Changed getOption to getGlobalPreference for skin; Changed getOption to getGlobalPreference for output of getUserOptionKey

tiffperumpail committed:
Changed getOption to getGlobalPreference for language

tiffperumpail committed:
Changed getOption to getGlobalPreference for language

tiffperumpail committed:
Changed getOption to getGlobalPreference for searchAllNamespaces

tiffperumpail committed:
Changed getOption to getGlobalPreference for thumbsize

tiffperumpail committed:
Changed getOption to getGlobalPreference for editor and changed getOption to getGlobalPreference in comment description for userGetOptions function

tiffperumpail committed:
Chnaged getOption to getGlobalFlag for poweruser_admin, poweruser_frequent, and poweruser_lifetime

tiffperumpail committed:
Changed getOption to getGlobalPreference for thumbsize

tiffperumpail committed:
Changed getOption to getGlobalPreference for rememberpassword; Changed getOption to getGlobalPreference to language

Commit:
Changed getOption to getGlobalPreference for enableGoSearch in GlobalNavigationController

Commit:
Changed getBoolOption to getGlobalPreference for watchlistdigesst, unsubscribed in GlobalWatchlistBot

Commit:
Changed getBoolOption to getGlobalPreference for watchlistdigesst, unsubscribed in GlobalWatchlistHooks

Commit:
Changed getOption to getGlobalPreference for imageReviewSort in ImageReviewSpecialController

Commit:
Changed getOption to getGlobalAttribute for NotConfirmedEmail in JSVariables

Commit:
Changed setOption to setGlobalPreference for imageReviewSort in ImageReviewSpecialController

Commit:
Changed getOption to getGlobalAttribute for LicensedVideoSwap_visited(List|Date) in LicensedVideoSwapHelper

Commit:
Changed getOption to getGlobalPreference for skin in LicensedVideoSwapHooksHelper

Commit:
Changed set/getOption to set/getGlobalAttribute for LicensedVideoSwap_visited(List|Date) in LicensedVideoSwapSpecialController

Commit:
Changed getOption to getGlobalPreference for disablelinksuggest in LinkSuggestHooks

Commit:
Changed getOption to getGlobalAttribute/Flag for disabled-user-email, new_email, allowAdoption in LookupUser

Commit:
add setGlobalFlag function to User without implementation

Commit:
change setGlobalAttribute for disabled-user-email, disabled_date in Wikia.php. Set GlobalFlag

Commit:
change getOption to getGlobalPreference for disablelinksuggest in MiniEditorController

Commit:
change getOption to getGlobalPreference for myhomedefaultview, userlandingpage, hidepersonalachievements in MyHome.php

Commit:
change getOption to getGlobalPreference for nocache in AjaxResponse

Commit:
change getOption to getGlobalPreference, getGlobalAttribute for watchlisttoken, and custom preference in ApiBase

Commit:
change getOption to getGlobalPreference, getGlobalAttribute for minordefault, and custom preference in ApiEditPage

Commit:
change getOption to getGlobalAttribute for gender in ApiQueryUsers

Commit:
change getOption to getGlobalPreference for getGlobalPreference, watchdeletion in Article.php

Commit:
change getOption to getGlobalPreference for usenewrc  in ChangesList.php

Commit:
change getOption to getGlobalPreference for skin, language in RequestContext.php

Commit:
change getBoolOption to getGlobalPreference for diffonly in DifferenceEngine.php

Commit:
change getOption to getGlobalPreference for previewonfirst in EditPage.php

Commit:
change getOption to getGlobalPreference in EmailNotification

Commit:
change setOption to setGlobalPreference in Unsubscribe.body.php

drsnyder committed:
getOption changes in extensions/wikia/CategorySelect/CategorySelectHelper.class.php

disablecategoryselect

drsnyder committed:
getOption changes related to request-closure

Changes in:
 - extensions/wikia/CloseMyAccount/CloseMyAccountHelper.class.php
 - extensions/wikia/CloseMyAccount/maintenance/CloseMyAccountMaintenance.php

 Options affected:
  * flag: requested-closure
	* flag: disabled-by-user-request
	* attribute: requested-closure-date
	* preference: language

Note that the maintenance script inverted search for "options" may not
be supported in the future.

/cc @grunny

drsnyder committed:
getOption changes in extensions/wikia/CreateNewWiki/CreateNewWikiController.class.php

language

drsnyder committed:
getOption changes in  extensions/wikia/CreatePage/CreatePage.php

createpagepopupdisabled

drsnyder committed:
getOption changes in extensions/wikia/CreatePage/monobook/SpecialEditPage.class.php

disablecategoryselect

ArturKlajnerok committed:
getOption changed in includes/specials/SpecialContributions.php

ArturKlajnerok committed:
getOption changed in includes/SkinTemplate.php

ArturKlajnerok committed:
getOption changed in includes/SkinLegacy.php

ArturKlajnerok committed:
getOption changed in includes/ProtectionForm.php

ArturKlajnerok committed:
getOption changed in includes/FileDeleteForm.php

ArturKlajnerok committed:
getOption changed in includes/ImagePage.php

Commit:
Change getOption to getGlobalPreference in GlobalWatchListTest

ArturKlajnerok committed:
getOption changed in includes/Message.php

drsnyder committed:
getOption changes in extensions/wikia/EditAccount/SpecialEditAccount_body.php

requested-closure,new_email,unsubscribed,disabled,disabled_date,AllowAdoption,NotConfirmedSignup,SignedUpOnWiki

ArturKlajnerok committed:
getOption changed in includes/search/SearchEngine.php

ArturKlajnerok committed:
getOption changed in includes/parser/CoreParserFunctions.php

ArturKlajnerok committed:
getOption changed in includes/parser/Parser.php

drsnyder committed:
getOption changes in extensions/wikia/EditorPreference/EditorPreference.class.php

editor

drsnyder committed:
getOption changes in extensions/wikia/EditPageLayout/EditPageLayoutHelper.class.php

disablesyntaxhighlighting,editwidth

ArturKlajnerok committed:
getOption changed in includes/parser/ParserOptions.php

ArturKlajnerok committed:
getOption changed in extensions/wikia/FounderEmails/events/FounderEmailsCompleteDigestEvent.class.php

ArturKlajnerok committed:
getLocalPreference fixed in extensions/wikia/FounderEmails/events/FounderEmailsDaysPassedEvent.class.php

tiffperumpail committed:
Changed getOption to getGlobalFlag for NotConfirmedSignup

tiffperumpail committed:
Merge branch 'SERVICES-469' of https://github.com/Wikia/app into SERVICES-469

drsnyder committed:
getOption changes in extensions/wikia/EditPageLayout/models/EditorUserPropertiesHandler.class.php

Changed the EditorUserPropertiesHandler class to use setLocalAttributes
to differentiate it from preferences. As part of this I also changed the
handling of the EduNote methods to use set/getLocalAttribute.

Attributes affected EditorMainPageNotificationHidden.

/cc @sebastian-wikia

drsnyder committed:
Add setLocalAttribute

drsnyder committed:
getOption changes in extensions/wikia/Email/EmailController.class.php

language, unsubscribed

drsnyder committed:
getOption changes in extensions/wikia/ExactTargetUpdates/hooks/ExactTargetUserHooksHelper.php

marketingallowed, unsubscribed, language

drsnyder committed:
getOption changes in extensions/wikia/FacebookClient/FacebookClientController.class.php

fbFromExist

drsnyder committed:
getOption changes in extensions/wikia/Follow/FollowedPagesSpecial.php

hidefollowedpages

drsnyder committed:
getOption changes in extensions/wikia/Follow/FollowHelper.class.php

hidefollowedpages, enotifwallthread

drsnyder committed:
getOption changes in extensions/wikia/Forum/ForumController.class.php

forum_sort_

ArturKlajnerok committed:
getOption changed in extensions/wikia/FounderEmails/events/FounderEmailsDaysPassedEvent.class.php

ArturKlajnerok committed:
Merge branch 'SERVICES-469' of https://github.com/Wikia/app into SERVICES-469

drsnyder committed:
Fix categorization of unsubscribed in extensions/wikia/EditAccount/SpecialEditAccount_body.php

drsnyder commented on extensions/wikia/EditAccount/SpecialEditAccount_body.php:
```diff
@@ -219,10 +219,10 @@ public function execute( $par ) {
 					'userReg' => date( 'r', strtotime( $this->mUser->getRegistration() ) ),
-					'isUnsub' => $this->mUser->getOption('unsubscribed'),
-					'isDisabled' => $this->mUser->getOption('disabled'),
+					'isUnsub' => $this->mUser->getGlobalFlag('unsubscribed'),
```
@ArturKlajnerok fixed in 8a3e5fa4.

Commit:
update get/setOption calls in extensions/wikia/RecentChanges/RecentChangesFiltersStorage.class.php

ArturKlajnerok committed:
getOption changed in extensions/wikia/FounderEmails/events/FounderEmailsEditEvent.class.php

ArturKlajnerok committed:
getOption changed in extensions/wikia/FounderEmails/events/FounderEmailsRegisterEvent.class.php

ArturKlajnerok committed:
getOption changed in extensions/wikia/FounderEmails/events/FounderEmailsViewsDigestEvent.class.php

ArturKlajnerok committed:
Merge branch 'SERVICES-469' of https://github.com/Wikia/app into SERVICES-469

ArturKlajnerok committed:
getOption changed in extensions/wikia/FounderEmails/FounderEmails.class.php

ArturKlajnerok committed:
getOption changed in extensions/wikia/FounderEmails/convert.php

ArturKlajnerok committed:
setOption changed in extensions/Math/Math.hooks.php

Commit:
changing get/setOption calls in includes/Preferences.php

ArturKlajnerok committed:
getOption changed in extensions/SemanticWatchlist/SemanticWatchlist.hooks.php

ArturKlajnerok committed:
getOption changed in extensions/wikia/AchievementsII/Ach_setup.php

ArturKlajnerok committed:
getOption changed in extensions/wikia/AchievementsII/services/AchAjaxService.class.php

ArturKlajnerok committed:
Merge branch 'SERVICES-469' of https://github.com/Wikia/app into SERVICES-469

Commit:
change setOption to setGlobalPreference for rememberpassword in server.php

Commit:
change getOption to getGlobalPreference for upwiz_deflicense in SpecialUploadWizard.php

Commit:
change getOption to getGlobalPreference for watchdefault in SpecialAddVideo.php

Commit:
change getOption to getGlobalPreference for visualeditor-enable, visualeditor-betatempdisable in VisualEditor.hooks.php

Commit:
change getOption to getGlobalPreference for visualeditor-enable, visualeditor-betatempdisable in VisualEditor.hooks.php (old)

Commit:
change setOption to setGlobalPreference for marketingallowed, language in AutoFollowHooksTest.php

Commit:
change setOption to setGlobalPreference for adoptionmails  in AutomaticWikiAdoptionHelper.php

Commit:
change getOption to getGlobalFlag for disabled  in AutomaticWikiAdoptionHelper.php

Commit:
change getOption to getGlobalFlag for disabled  in FlagClosedAccounts.php

Commit:
change setOption to setGlobalFlag for requested-rename, requested-closure  inSpecialContact.body.php

Commit:
change getBoolOption to getGlobalFlag fordisabled  in FacebookSignupController.class.php

Commit:
change setOption to setGlobalAttribute for gender in UserLoginFacebbokForm.class.php

Commit:
change setOption to setGlobalAttribute, setGlobalFlag for UserLoginSpecialController attributes and flags in UserLoginForm.class.php

Commit:
change setOption to setGlobalFlag for cr_mailed in UserLoginHelper.class.php

Commit:
change setOption to setGlobalAttribute for new_email in UserLoginHooksHelper.class.php

Commit:
change setOption to setGlobalPreference for rememberpassword in UserLoginSpecialController.class.php

Commit:
change setOption to setGlobalAttribute for SignupRedirect in UserSignupSpecialController.class.php

Commit:
change setOption to setGlobalAttribute for SignupRedirect, new_email in WikiaConfirmEmailSpecialController.class.php

Commit:
change setOption to setGlobalAttribute for avatar in MastHead.class.php

Commit:
change setOption to setGlobalAttribute for mastheadoption in UserIdentityBox.class.php

Commit:
change setOption to setGlobalAttribute for avatar, avatar_rev in UserProfilePageController.class.php

Commit:
change setOption to setGlobalAttribute, setGlobalFlag for renameData, disabled in RenameUserProcess.class.php

Commit:
change setOption to setGlobalFlag for wasRenamed in UserRenameTask.class.php

Commit:
change setOption to setGlobalPreference for wall_sort_ in WallBaseController.class.php

Commit:
change setOption to setGlobalPreference for wall_sort_ in WallController.class.php

Commit:
change getBoolOption,getOption to getGlobalPreference for unsubscribed, enotifwallthread  in WallNotifications.class.php

Commit:
change setOption to setGlobalPreference for widget_bookmark_pages in WidgetBookmark.php

Commit:
change setOption to setGlobalPreference for rememberpassword in WikiaApiAjaxLogin.php

Commit:
change getOption to getGlobalPreference for diffonly in CategoryPage.php

Commit:
change getOption to getGlobalPreference for EditPage preferences in EditPage.php

Commit:
change getOption to getGlobalPreference for externaldiff, externaleditor in ExternalEdit.php

Commit:
change getOption to getGlobalPreference for cols, rows, nocache, disablesuggest, editsectiononrightclick in OutputPage.php

Commit:
change getOption to getGlobalPreference for rclimit in Pager.php

Commit:
change getBoolOption to getGlobalPreference for showhiddencats in Skin.php

Commit:
change getBoolOption to getGlobalPreference for norollbackdiff in RollbackAction.php

Commit:
change setOption to setGlobalPreference for rememberpassword in ApiLogin.php

Commit:
change setOption to setGlobalPreference for various option from ApiOptions.php

Commit:
change getBoolOption to getGlobalPreference forccmeonemails in SpecialEmailuser.php

Commit:
change getBoolOption, getOption to getGlobalPreference rclimit, newpageshidepatrolled in SpecialNewPages.php

Commit:
change setOption to setGlobalPreference usenewrc in SpecialRecentchanges.php

Commit:
remove third parameter

ArturKlajnerok committed:
getOption changed extensions/wikia/Helios/User.class.php

ArturKlajnerok committed:
setOption changed extensions/wikia/EditPageLayout/EditPageLayoutAjax.class.php

Commit:
Add a test for testCreateLocalOptionName

ArturKlajnerok committed:
getUserPropertyValue removed extensions/wikia/EditPageLayout/models/EditorUserPropertiesHandler.class.php

ArturKlajnerok committed:
Merge branch 'SERVICES-469' of https://github.com/Wikia/app into SERVICES-469

ArturKlajnerok committed:
getOption changed extensions/wikia/CreatePage/CreatePage.php

ArturKlajnerok committed:
setOption changed extensions/wikia/SkinChooser/SkinChooser.class.php

ArturKlajnerok commented on includes/User.php:
```diff
@@ -2416,6 +2416,177 @@ public function getIntOption( $oname, $defaultOverride=0 ) {
+	 * @param char $sep the separator between the option and the id.
+	 * @return string
+	 */
+	public static function createLocalOptionName($option, $cityId = null, $sep = '-') {
```
Shouldn't this be a `private` function ?

drsnyder replied:
But then you can't test it (easily).

ArturKlajnerok committed:
setOption changed extensions/wikia/PowerUser/PowerUser.class.php

drsnyder committed:
getOption changes in includes/wikia/services/tests/AvatarServiceTest.php

avatar

drsnyder committed:
getOption changes in includes/wikia/services/ToolbarService.class.php

*-toolbar-contents,*-toolbar-promotions

ArturKlajnerok committed:
setOption changed includes/specials/SpecialUserlogin.php

drsnyder committed:
Remove savePropertyValue

I could not find any references to this function being called so I
removed it in order to address the getOption call site.

/cc @sebastian-wikia @nAndy

ArturKlajnerok committed:
fixed getGlobalAttribute to setGlobalAttribute includes/specials/SpecialUserlogin.php

ArturKlajnerok committed:
setOption changed includes/specials/SpecialWatchlist.php

ArturKlajnerok committed:
Merge branch 'SERVICES-469' of https://github.com/Wikia/app into SERVICES-469

Commit:
change get/set global preferences to use UserPreferences class, which uses the preference service

Commit:
Fix testNewFromTokenAuthorizationGranted test

drsnyder committed:
getOption changes in languages/LanguageConverter.php

nonconvertlink

drsnyder committed:
getOption to getGlobalPreference in maintenance/userOptions.inc

This could potentially change any preference. I changed it to be isolated to user preferences.

drsnyder committed:
getOption changes in maintenance/wikia/cronjobs/AutoFollow.php

autowatched-already

drsnyder committed:
getOption changes in maintenance/wikia/EnableMagCloud.php

widgets

drsnyder committed:
getOption changes in maintenance/wikia/EnableWysiwygSitewide.php

EnableWysiwyg

Commit:
set/load preferences from memcache

Commit:
use constant

drsnyder committed:
getOption changes in maintenance/wikia/migrateEditorPreference.php

editor

Commit:
remove hard type hint for deserialization

drsnyder committed:
getOption changes in maintenance/wikia/switchUserSkin.php

skin

drsnyder committed:
getOption changes in tests/phpunit/includes/specials/SpecialSearchTest.php

searchNS

drsnyder committed:
getOption changes in skins/oasis/modules/MyToolsController.class.php

myTools

drsnyder committed:
getOption changes in tests/phpunit/includes/LanguageConverterTest.php

variant

ArturKlajnerok committed:
deprecated notice in extensions/Vector/switchExperimentPrefs.php

Commit:
add input sanitization to setGlobalPreference and hook for getting a preference

Commit:
change getOption to getGlobalPreference for vector-simplesearch in Vector.php

Commit:
change getOption to getGlobalPreference for underline, highlightbroken in Simple.php

Commit:
change getOption to getGlobalPreference for smw-prefs-ask-options-tooltip-display in SMW_QuerySpecialPage.php

Commit:
change setOption to setGlobalPreference for toolbaroptions, setGlobalAttribute for Masthead options in UserPreferencesV2.php

Commit:
Merge branch 'dev' into SERVICES-469

Conflicts:
	extensions/wikia/FounderEmails/events/FounderEmailsCompleteDigestEvent.class.php
	extensions/wikia/FounderEmails/events/FounderEmailsDaysPassedEvent.class.php
	extensions/wikia/FounderEmails/events/FounderEmailsEditEvent.class.php
	extensions/wikia/FounderEmails/events/FounderEmailsRegisterEvent.class.php
	extensions/wikia/FounderEmails/events/FounderEmailsViewsDigestEvent.class.php
	extensions/wikia/UserLogin/UserLoginForm.class.php
	extensions/wikia/UserLogin/UserLoginHelper.class.php
	extensions/wikia/UserLogin/UserLoginSpecialController.class.php
	extensions/wikia/UserLogin/WikiaConfirmEmailSpecialController.class.php

Commit:
change getOption for getGlobal(Preference|attribute), for founderemails  optionns in FounderEmailsEditEvent.php

Commit:
change getBoolOption forgetGlobalPreference, for founderemails  options in FounderController.php

Commit:
deprecate getOption and setOption in User.php

Commit:
change setGlobalPreference to setLocalPreference for adoptionmails in AutomaticWikiAdoptionHelper

Commit:
change getGlobalPreference to getLocalPreference for founderemails-complete-digest, founderemails-joins in FounderController

Commit:
change getGlobalPreference to getLocalPreference for founderemails-complete-digest in FounderEmailsCompleteDigestEvent

Commit:
change getGlobalPreference to getLocalPreference for founderemails-complete-digest, founderemails-edits in FounderEmailsEditEvent

Commit:
change getGlobalPreference to getLocalPreference for founderemails-complete-digest, founderemails-views-digest in FounderEmailsViewsDigestEvent

sqreek commented:
[getOption](https://github.com/Wikia/app/pull/7648/files#diff-897fba1dd6b8edc3238762e35d0b7c71R2320) and [setOption](https://github.com/Wikia/app/pull/7648/files#diff-897fba1dd6b8edc3238762e35d0b7c71R2596) are deprecated now

Commit:
adding ignoreHidden to getGlobalPreference and $default/$ignoreHidden to getLocalPreference

Commit:
reorganizing code and add get/setOptionHelper

Commit:
Merge remote-tracking branch 'origin/SERVICES-469' into use-prefs-service

Conflicts:
	includes/User.php
	lib/Wikia/src/Persistence/User/PreferencePersistenceMySQL.php

Commit:
move preferences module and user preferences service to lib/wikia

Commit:
fixing errors after moving classes, adding hook for preference sets

Commit:
log when calling set/getoption

Commit:
adding tests for UserPreferences

Commit:
Merge remote-tracking branch 'origin/dev' into SERVICES-469

Commit:
Merge branch 'SERVICES-469' into use-prefs-service

Commit:
fixing tests in lib/Wikia/tests/Persistence/User/PreferencePersistenceMySQLTest.php

Commit:
"avatar" is listed as an attribute /cc @drsnyder

Commit:
can't have as part of user, since it will break memcache due to database connection in object

Commit:
updating static function name

ArturKlajnerok committed:
casting bool for all getBoolOption replacements

garthwebb commented on includes/User.php:
```diff
@@ -2416,11 +2417,183 @@ public function getIntOption( $oname, $defaultOverride=0 ) {
+		global $wgCityId;
+		if (!isset($cityId)) {
+			$cityId = $wgCityId;
+		}
```
@drsnyder I think I'm missing something ... shouldn't there be more code here?  Something to actually get the preference value?

drsnyder replied:
@garthwebb it's magic! Actually that work is happening over [here](https://github.com/Wikia/app/pull/7723). That branch was taken off of this one and will be merged in here before we merge to dev. So yes, when we are done something will actually get the preferences :).

Commit:
add language to format for url

Commit:
don't save anon prefs, change to flag

Commit:
updating unit tests

Commit:
updating test

Commit:
fixing sprintf

ArturKlajnerok committed:
fixed formatting

Commit:
updating test

Commit:
explicitly set up getGlobalFlag for failing test case

Commit:
make getOptionHelper private

Commit:
remove redundant code /cc @drsnyder

Commit:
getOption -> getGlobalFlag for the user login test

Commit:
separating to function /cc @drsnyder

Commit:
defaulting to false /cc @drsnyder

Commit:
updating mock and re-enabling to make sure jenkins tests pass

Commit:
re-disabling flag

Commit:
re-add use of ignoreHidden flag

nmonterroso committed:
Merge pull request #7723 from Wikia/use-prefs-service

Use prefs service

ArturKlajnerok commented:
Lgtm :+1:

Commit:
updating getoption/setoption functions in includes/User.php
